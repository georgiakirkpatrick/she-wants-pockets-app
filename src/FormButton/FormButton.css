* {
    box-sizing: border-box;
    padding: 0;
    margin: 0;
}

.FormButton {
    margin: 10px;
    border-radius: 15px;
    width: 90%;
    background-color: rgb(71, 48, 92);
    color: #fff;
    font-size: 16px;
    font-weight: 600;
    padding: 15px 32px;
    display: inline-block;
    text-align: center;
}

.disabled {
    opacity: 0.6;
    cursor: not-allowed;
}

.FormButton:active {
    background-color:  rgb(41, 5, 75);
    color: white;
}