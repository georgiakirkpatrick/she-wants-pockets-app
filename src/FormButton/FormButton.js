import React from 'react'
import './FormButton.css'

const FormButton = props => {
    return (
        <button className='FormButton' type={props.type} onClick={props.click}>
            {props.buttonText}
        </button>
    )
}

FormButton.defaultProps = {
    type: 'button'
}

export default FormButton