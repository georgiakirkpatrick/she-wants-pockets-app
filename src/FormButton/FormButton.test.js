import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import FormButton from './FormButton';

describe('FormButton', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<FormButton />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it ('render the UI as expected', () => {
        const tree = renderer
            .create(<FormButton />)
            .toJSON();
        expect(tree).toMatchSnapshot()
    })
})