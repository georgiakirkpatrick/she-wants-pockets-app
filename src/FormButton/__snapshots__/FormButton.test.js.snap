// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormButton render the UI as expected 1`] = `
<button
  className="FormButton"
  type="button"
/>
`;
